src/App.jsx

```diff
@@ -3,21 +3,34 @@ import './App.css';
 import TodoList from './TodoList';
 import AddTodoForm from './AddTodoForm';
 
-const useSemiPersistentState = () => {
+function App() {
   const [todoList, setTodoList] = React.useState(
-    JSON.parse(localStorage.getItem('savedTodoList') || '[]')
+    JSON.parse(localStorage.getItem('savedTodoList')) || []
   );
 
-  React.useEffect(() => {
-    const savedTodoList = JSON.stringify(todoList);
-    localStorage.setItem('savedTodoList', savedTodoList);
-  }, [todoList]);
+  const [isLoading, setIsLoading] = React.useState(true);
 
-  return [todoList, setTodoList];
-};
+  React.useEffect(() => {
+    new Promise((resolve, reject) => {
+      setTimeout(() => {
+        resolve({
+          data: {
+            todoList: JSON.parse(localStorage.getItem('savedTodoList') || []),
```

Comment:
This line is producing an error, currently is trying to parse an already parsed element ( an array ) `[ ]`, to make it work you could put that array between quotes like:
`            todoList: JSON.parse(localStorage.getItem('savedTodoList') || '[]'),
`
or only parse the left operand: 
`            todoList: JSON.parse(localStorage.getItem('savedTodoList')) || [],
`

⚠️ **Important note!** ⚠️ 
That line will throw the following error only when there're no todos saved in local storage, if you have todos saved in your local storage you won't be able to notice the error since the left operand in the logical OR condition will give a thruthy value, thus the JSON.parse([]) won't be evaluated.
```
VM647:1 Uncaught SyntaxError: Unexpected end of JSON input
    at JSON.parse (<anonymous>)
    at App.jsx:18:28
```

Reply:
Thank you @gsvidal! I will make the requested changes and will add it to the next commit.